### ISSUE TITLE
Compatibility with Cobblemon

### ISSUE THREAD

**Issue by CubeMaster111** [Xalcon/TorchMaster#227]
Hi,

it appears Torchmaster is unable to prevent Cobblemon mobs from spawning, likely due to some custom spawning system on their side.
To my knowledge the mod spawns a base mob (cobblemon:pokemon) which it then loads with data. 
Adding the base mob to the blacklist of both the megatorch and the dreadlamp and setting aggressive spawn checks to true does not prevent the spawn of cobblemon mobs.

This was tried on
Minecraft 1.20.1
Forge 47.2.0
Torchmaster 20.1.5
Cobblemon 1.4.0

**Comment by Xalcon** [comment]
This is usually a problem with the mod, not with torchmaster. Pretty much all pokemon style mods have this issue because they dont utilize spawn events. I can look into it, but I'm not sure if it is possible to do this from my side.

**Comment by CubeMaster111** [comment]
Thanks for looking into it, even if it doesnt end up working
Happy xmas

**Comment by TLATER** [comment]
FYI, mostly for others who stumble upon this looking for a solution like I did, the cobblemon mod at least appears to plan adding this type of functionality in the future (source: chatter on their discourse).

For now there is a third party addon for "unimplemented items" that adds a repel block which does more or less what torchmaster does, except for the spawns from cobblemon: https://modrinth.com/datapack/cobblemon-unimplemented-items

Unobtainable by default, but you can easily add a datapack recipe for it, or use crafttweaker or whatnot.

No idea if this makes this particular issue unnecessary, I guess compatibility with torchmaster would still be nice for your one-stop-shop to preventing mob spawns, but I also imagine that'd just cause unwanted churn in this repo [given it does need to interact with a mod-specific spawning mechanism](https://github.com/timinc-cobble/cobblemon-unimplemented-items-1.4-fabric/blob/main/src/main/java/us/timinc/mc/cobblemon/unimplementeditems/mixins/AreaSpawnerMixin.java).

**Comment by SigmaFlame** [comment]
Is this issue fixed on latest version?
Cobblemon still does not have any way to prevent their pokemon spawn in 1.21.1.

And all solutions are for Fabric only.
Desperately searching a solution for NeoForge.

**Comment by Xalcon** [comment]
> Is this issue fixed on latest version? Cobblemon still does not have any way to prevent their pokemon spawn in 1.21.1.
> 
> And all solutions are for Fabric only. Desperately searching a solution for NeoForge.

I'll not add cobblemon support to torchmaster itself. Cobblemon is written in kotlin and there are a few technicle hurdles getting that to work in my multiloader setup.

I did, however, create a compat mod. Not sure if I will keep it up to date, we will see. I personally have no interest in cobblemon myself, but I wanted to see how difficult it is to add support. Anyway, I've uploaded it as a dedicated project to Curseforge. It may take a while until curseforge approves it and it becomes visible. There is only support for Minecraft 1.21.1 via NeoForge.

I've only tested it for a few minutes.

https://legacy.curseforge.com/minecraft/mc-mods/torchmaster-cobblemon-compat
